src/lib/email.ts

```diff
@@ -236,51 +161,80 @@ ${siteConfig.developer.name}
   }
 }
 
-// Fallback console logger for development
+class SMTPEmailService implements EmailService {
+  private transporter;
+
+  constructor() {
+    this.transporter = nodemailer.createTransport({
+      host: import.meta.env.SMTP_HOST,
+      port: parseInt(import.meta.env.SMTP_PORT || "587"),
+      secure: import.meta.env.SMTP_SECURE === "true",
+      auth: {
+        user: import.meta.env.SMTP_USER,
+        pass: import.meta.env.SMTP_PASS,
+      },
+    });
+  }
+
+  isConfigured(): boolean {
+    return !!this.transporter;
+  }
+
+  async sendDownloadEmail(purchase: PurchaseData): Promise<void> {
+    const expiryDate = new Date(purchase.downloadExpires).toLocaleDateString();
+
+    await this.transporter.sendMail({
+      from: `${siteConfig.developer.name} <${siteConfig.email.from_downloads}>`,
+      to: purchase.customerEmail,
+      subject: `Your ${purchase.gameTitle} download is ready!`,
+      html: new ResendEmailService().generateDownloadEmailHTML(purchase, expiryDate),
```

Comment:
You could move those html generators to a util function somewhere so that it's not tied to the resend provider as it will be used by all providers

Reply:
Got it, I'll move these to the `utils.ts` file.